src/ucp/api/ucp.h

```diff
@@ -4518,12 +4520,24 @@ typedef struct ucp_ep_attr {
      * this name.
      */
     char     name[UCP_ENTITY_NAME_MAX];
+
+    /**
+     * Local socket address for this endpoint. Valid only for endpoints created 
+     * by connecting to a socket address.
+     */
+    struct sockaddr_storage local_sockaddr;
```

Comment:
what happens if socket is not connected to a socket address ?

Reply:
@shamisp  Return UCS_ERR_NOT_CONNECTED

Reply:
So the API is valid for all endpoint and if the socket was not connected using socket cm  UCS_ERR_NOT_CONNECTED is not returned. This has to be properly documented.

Reply:
@shamisp I guess there is some misunderstanding here. We always use socket address to connect. But the EP's status can have two cases:
1) If the EP is connected, then the API is valid to get local sockaddr and remote sockaddr as documented above, and return UCS_OK.
2) If the EP is not connected, then return UCS_ERR_NOT_CONNECTED as the error status.

I added this description according to @yosefe 's comment https://github.com/openucx/ucx/pull/7020#discussion_r664354439.
@yosefe Please correct me if I made any misunderstanding about the description.

Reply:
there are cases when ep is connecting to worker address (not socket address) so there is no CM lane and we are returning UCS_ERR_NOT_CONNECTED. That's why we wrote "Valid only for endpoints created by connecting to a socket address." Perhaps should add "If this field is specified for an endpoint not connected to a socket address, the behavior is undefined" or ".. UCS_ERR_NOT_CONNECTED will be returned"

Reply:
@shamisp @yosefe Added. Thanks for the comments!

Reply:
@yosefe "behavior is undefined" is the worse ...

Reply:
@shamisp  I added "If this field is specified for an endpoint not connected to a socket address, UCS_ERR_NOT_CONNECTED will be returned."

Reply:
@shamisp Could please approve this PR? Let me know please if you have any more comments. Thanks!